개별 채널 페이지에서 참가자 목록 관리 API 호출

srngch opened the pull request:
## 작업 내용

- closes #236

## 리뷰어에게
- [x] 저도 채널 생성한 회원의 role이 owner로 수정된 후 리뷰 요청할 예정입니다~!

srngch committed:
add[FE]: Add api to get messages in channel

srngch committed:
add[FE]: Add api to set user role and status in channel

- PATCH `/channels/:channelId/users/:userId/role`
- PATCH `/channels/:channelId/users/:userId/status`
- Add `value` prop to Button component

hhkim0729 commented on frontend/src/api/api.v1.ts:
```diff
@@ -269,3 +269,41 @@ export const patchChannelSetting = async (channelForm: ChannelFormType) => {
+}: {
+  channelId: number;
+  userId: number;
+  role: 'admin' | 'normal';
```
RoleType을 활용하는 것이 좋아 보입니다~!

hhkim0729 commented on frontend/src/api/api.v1.ts:
```diff
@@ -269,3 +269,41 @@ export const patchChannelSetting = async (channelForm: ChannelFormType) => {
+}: {
+  channelId: number;
+  userId: number;
+  status: 'kick' | 'ban' | 'mute';
```
이것도 타입을 하나 만드는 게 좋을 것 같아요!

hhkim0729 commented on frontend/src/components/molecule/ChannelUserItem.tsx:
```diff
@@ -4,19 +4,39 @@ import { ChannelUserType, RoleType } from 'types/channelUserType';
 import Button from 'components/atoms/Button';
 
 interface Props {
+  changeRole: (arg: any) => void;
+  changeStatus: (arg: any) => void;
```
any일 수밖에 없는 걸까요..

srngch replied:
리팩토링하면서 변경..!

hhkim0729 commented on frontend/src/pages/ChannelPage.tsx:
```diff
@@ -55,6 +57,44 @@ export default function ChannelPage() {
 
+  const patchChannelUserRoleMutation = useMutation<any, AxiosError, any>({
+    mutationFn: patchChannelUserRole,
+    onSuccess: () => getChannelQuery.refetch(),
```
오호 복잡하게 할 필요 없이 쿼리가 refetch 함수를 제공하는군요.

onSuccess, onError 등을 useMutation에서 처리할지 mutate에서 할지 고민하다가 저는 mutate에서 하는 게 쿼리 실행 후에 어떤 일이 일어날지가 명시적이라 판단했는데, 사용하는 곳이 한 곳이라면 useMutation에서 해도 문제 없을 것 같네요. 다만 이것도 리팩토링 하면서 맞추는 것이 좋아 보입니다.

저 타입 명시도 계속 any를 보게 돼서 마음은 불편한데 AxiosError 쓰려고 하시는 거죠..?
근데 문제는 저렇게 받아와서 message를 출력해도 axios에서 기본으로 제공하는 메시지가 나오지 우리가 설정한 메시지가 나오지 않아서 정말 이게 필요한 것인지 고민을 또 해봐야 할 것 같네요...

hhkim0729 reviewed (dismissed):
전체적으로 반복되는 string 값 사용을 타입으로 대체하는 것이 좋아 보입니다.
any도 가능하면 줄이고 싶네유...

srngch committed:
refactor[FE]: Add types

- Add ChannelUserRoleType
- Add ChannelUserStatusType
- Rename object RoleType to UserRole
